lib/getTrainingParameters.m

```diff
@@ -138,12 +210,53 @@
     cfg.soundTrackBeat{cfg.isTrackIdx(tracki)} = seqBeat; 
 
     % get window duration in seconds
-    cfg.winDur(cfg.isTrackIdx(tracki)) = cfg.nCyclesPerWin * ...
-                                         cfg.cuePeriodGrid(cfg.isTrackIdx(tracki)) * ...
-                                         beatPeriod; 
+    cfg.winDur(cfg.isTrackIdx(tracki)) = cfg.nCyclesPerWin(cfg.isTrackIdx(tracki)) * beatPeriod; 
     
 end
 
+%% Grahn(2007) patterns
+
+cfg.grahn.fs = cfg.fs; 
+
+% read from txt files
+grahnPatSimple = loadIOIRatiosFromTxt(fullfile('stimuli','Grahn2007_simple.txt')); 
+grahnPatComplex = loadIOIRatiosFromTxt(fullfile('stimuli','Grahn2007_complex.txt')); 
+
+% get different metrics of the patterns
+cfg.grahn.patternSimple = getPatternInfo(grahnPatSimple, 'simple',cfg); 
+cfg.grahn.patternComplex = getPatternInfo(grahnPatComplex, 'complex', cfg); 
+
+% total sound duration proportion to gridIOI _/```\_  
+cfg.grahn.soundDurProp = 1; % 100% of gridIOI
+% onset ramp duration                       _/     
+cfg.grahn.eventRampon = 0.05; % 5% of gridIOI 
+% offset ramp duration                             \_ 
+cfg.grahn.eventRampoff  = 0.020; % 10% of gridIOI
+
+% the grid interval can vary across steps or segments (gridIOI selected 
+% randomly from a set of possible values for each new step or segment) 
+cfg.grahn.gridIOI = 0.190; 
+
+% construct pattern (smallest item in sequence)
+cfg.grahn.nGridPoints = 12; % length(pat_complex(1).pattern)
+
+cfg.grahn.interPatternInterval = cfg.grahn.nGridPoints * cfg.grahn.gridIOI; 
+
+% construct pitch features of the stimulus 
+% the pitch (F0) of the tones making up the patterns can vary 
+% (it can be selected randomly from a set of possible values)
+cfg.grahn.minF0 	= 350; % minimum possible F0
+cfg.grahn.maxF0 	= 900; % maximum possible F0
+cfg.grahn.nF0       = 5; % number of unique F0-values between the limits
+cfg.grahn.F0s       = logspace(log10(cfg.grahn.minF0),log10(cfg.grahn.maxF0),cfg.grahn.nF0); 
+
+% use the requested gain of each tone to adjust the base amplitude
+cfg.grahn.F0sAmpGain = equalizePureTones(cfg.grahn.F0s,[], []);
+cfg.grahn.F0sAmps = 1/sqrt(2) * maxAllowedRms * cfg.grahn.F0sAmpGain; 
```

Comment:
these are the common with `mainExp getParams`. Maybe I can call the `mainExp` `getParams` here for these parameters? So there'll be no duplication of calling them. And also 1 place to make modification in the scripts. What do you think?

Reply:
It's a good idea, and I've been considering this. 

But I think we might appreciate a little bit of independence for the tap trainer, not having our hands tied by whatever changes we make in the main experiment parameters. I know we could call getMainExpParameters and then change the cfg fields we need, but I feel like this way it's a bit more clear? 

Also, there's a lot of parameters that getMainExpParameters returns, which are not used for the tapTrainer at all. So I'm afraid these parameters would just make the tapTrainer log hard to understand. I've made the stimulus construction here a bit simpler, so you can only do one-category sequences, and there's special part of makeStimTrain that takes care of it, so we stay away from patching-up and tweaking parameters from getMainExpParameters we don't need here, just so that we can call the same functions. I hope the way I did makes it more transparent...but perhaps it does the opposite...:DD